core/Map.carp

```diff
@@ -156,6 +178,43 @@
           (let [e (Array.nth entries j)]
             (f (Pair.a e) (Pair.b e)))))))
 
+  (doc endo-map "Transform values of the given map in place. f gets two arguments, key and value, and should return new value")
+  (defn endo-map [f m]
+    (do
+      (for [i 0 @(n-buckets &m)]
+        (let [bucket (Array.nth (buckets &m) i)
+              len (Array.length (Bucket.entries bucket))
+              entries (Bucket.entries bucket)]
+          (for [j 0 len]
+            (let [e (Array.nth entries j)]
+              (Array.aset! entries j (Pair.init @(Pair.a e)
+                                                (f (Pair.a e) (Pair.b e))))))))
```

Comment:
I'd love to do something like this:
```
(Array.endo-map (fn [p] (Pair.update-b p f)) entries)
```
but that too is broken by #347. f would have to be a ref, and closing over function refs is broken

Reply:
Yeah, definitely that looks like *the* thing to be able to do! Sorry about lambdas being even more broken than I thought... ^^